src/core/config/openthread-core-config-check.h

```diff
@@ -518,6 +518,24 @@
 
 #endif // OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE
 
+#if OPENTHREAD_CONFIG_SRP_SERVER_ENABLE
```

Comment:
It's good to have thee checks for config.

My suggestion is to add them in `srp_server.hpp` header itself. I feel it may be better place for them (close to the definition and implementation). This header file started as a place to track configs that are removed or renamed.

Thoughts?

Reply:
I see 1.2 feature checks there, so followed it. But having individual modules checks by themselves sounds good to me. @jwhui @simonlingoogle Thoughts?

Reply:
Moved the check to `srp_server.hpp`.